sale_order_partial_advance/wizard/sale_line_invoice.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#     This file is part of sale_order_partial_advance,
+#     an Odoo module.
+#
+#     Copyright (c) 2015 ACSONE SA/NV (<http://acsone.eu>)
+#
+#     sale_order_partial_advance is free software:
+#     you can redistribute it and/or modify it under the terms of the GNU
+#     Affero General Public License as published by the Free Software
+#     Foundation,either version 3 of the License, or (at your option) any
+#     later version.
+#
+#     sale_order_partial_advance is distributed
+#     in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+#     even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+#     PURPOSE.  See the GNU Affero General Public License for more details.
+#
+#     You should have received a copy of the GNU Affero General Public License
+#     along with sale_order_partial_advance.
+#     If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models, api, fields
+from openerp.tools.translate import _
+
+
+class SaleOrderLineMakeInvoice(models.TransientModel):
+    _inherit = 'sale.order.line.make.invoice'
+
+    order_ids = fields.One2many(string='Sale Orders',
+                                comodel_name='sale.order.make.invoice',
+                                inverse_name='wizard_id')
+
+    @api.model
+    def default_get(self, fields_list):
+        defaults = super(SaleOrderLineMakeInvoice,
+                         self).default_get(fields_list)
+        lines = self.env['sale.order.line'].browse(
+            self.env.context.get('active_ids', []))
+        orders = set([line.order_id for line in lines
+                      if line.order_id.advance_amount_available > 0])
+        if orders:
+            order_ids = []
+            for order in orders:
+                all_lines = True
+                res = [line.id for line in order.order_line
+                       if line not in lines and not line.invoiced]
+                if res:
+                    all_lines = False
+                to_use = order.advance_amount_available if all_lines else 0.0
+                order_ids.append(
+                    (0, 0, {'wizard_id': self.id,
+                            'order_id': order.id,
+                            'advance_amount_available':
+                            order.advance_amount_available,
+                            'advance_amount_to_use': to_use,
+                            'all_lines': all_lines}))
+            defaults['order_ids'] = order_ids
+        return defaults
+
+    @api.model
+    def _prepare_invoice(self, order, lines):
+        wizard = order.env.context.get('instance')
```

Comment:
a comment above the line would be welcome to keep a trace of the motivation behind this kind of "hack" :smirk: